### PR TITLE
planner: enable inline projection for hash join

### PR DESCRIPTION
**Pull request by SunRunAway**
### What problem does this PR solve? 

Hash Join for #14428

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - None

Side effects

 - None

Related changes

 - None

### CODE REVIEW

File: planner/core/task.go [hunk 1]
```diff
@@ -522,7 +522,6 @@ func (p *PhysicalHashJoin) attach2Task(tasks ...task) task {
 	lTask := finishCopTask(p.ctx, tasks[0].copy())
 	rTask := finishCopTask(p.ctx, tasks[1].copy())
 	p.SetChildren(lTask.plan(), rTask.plan())
-	p.schema = BuildPhysicalJoinSchema(p.JoinType, p)
```
Comment (francis0407) [inline planner/core/task.go]: Shall we also remove this in `planner/cascades/implementation/join.go/HashJoinImpl.AttachChildren`?

Comment (SunRunAway) [reply planner/core/task.go]: done.

File: planner/cascades/integration_test.go [hunk 2]
```diff
@@ -292,3 +292,27 @@ func (s *testIntegrationSuite) TestTopN(c *C) {
 		tk.MustQuery(sql).Check(testkit.Rows(output[i].Result...))
 	}
 }
+
+func (s *testIntegrationSuite) TestInlineProjection(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("drop table if exists t1, t2;")
+	tk.MustExec("create table t1(a bigint, b bigint, index idx_a(a), index idx_b(b));")
+	tk.MustExec("create table t2(a bigint, b bigint, index idx_a(a), index idx_b(b));")
```
Comment (lzmhhh123) [inline planner/cascades/integration_test.go]: Would you please add some data to the tables and make results not null?

Comment (SunRunAway) [reply planner/cascades/integration_test.go]: done

**Comment by SunRunAway** [comment]
@lzmhhh123 @francis0407 
comments are addressed, and also fix a bug with outer join in https://github.com/pingcap/tidb/pull/14783/files#diff-85c322a7979869efbabdffba4dac51ff

PTAL again, Thanks.

**Pull request merged**